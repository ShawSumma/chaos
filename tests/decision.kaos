nil def hey()
    print "hey"
end

nil def hoy()
    print "hoy"
end

num def add(num x, num y)
    num z = x + y
    bool a = true
    return z
end { a && true : hey() }

print add(3, 5)

num def sub(num x, num y)
    num z = x - y
    bool a = true
    return z
end { a && false : hey(), default : hoy() }

print sub(5, 3)

num def f1()
    num a = 101
    return a
end

num def f2()
    num b = 102
    return b
end

num def f3(num x, num y)
    num z = x + y
    bool a = true
end { a && false : f1(), default : f2() }

print f3(3, 5)
