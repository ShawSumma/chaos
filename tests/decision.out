hey
8
hoy
2
102
